src/darwin/Framework/CHIPTests/MTROTAProviderTests.m

```diff
@@ -864,7 +898,437 @@ - (void)test005_DoBDXTransferAllowUpdateRequest
+
+    // Check if the ota raw image exists at kOtaRawImagePath1 and kOtaRawImagePath2
+    XCTAssertTrue([[NSFileManager defaultManager] fileExistsAtPath:otaRawImagePath1]);
+    XCTAssertTrue([[NSFileManager defaultManager] fileExistsAtPath:otaRawImagePath2]);
+
+    XCTestExpectation * queryExpectation1 = [self expectationWithDescription:@"handleQueryImageForNodeID called for device1"];
+    XCTestExpectation * queryExpectation2 =
+        [self expectationWithDescription:@"handleQueryImageForNodeID called for device2 - busy"];
+    XCTestExpectation * queryExpectation3 =
+        [self expectationWithDescription:@"handleQueryImageForNodeID called for device2 - available"];
+
+    XCTestExpectation * bdxBeginExpectation1 =
+        [self expectationWithDescription:@"handleBDXTransferSessionBeginForNodeID called for device1"];
+    XCTestExpectation * bdxBeginExpectation2 =
+        [self expectationWithDescription:@"handleBDXTransferSessionBeginForNodeID called for device2"];
+
+    XCTestExpectation * bdxQueryExpectation1 = [self expectationWithDescription:@"handleBDXQueryForNodeID called for device1"];
+    XCTestExpectation * bdxQueryExpectation2 = [self expectationWithDescription:@"handleBDXQueryForNodeID called for device2"];
+
+    XCTestExpectation * bdxEndExpectation1 =
+        [self expectationWithDescription:@"handleBDXTransferSessionEndForNodeID called for device1"];
+    XCTestExpectation * bdxEndExpectation2 =
+        [self expectationWithDescription:@"handleBDXTransferSessionEndForNodeID called for device2"];
+
+    XCTestExpectation * applyUpdateRequestExpectation1 =
+        [self expectationWithDescription:@"handleApplyUpdateRequestForNodeID called for device1"];
+    XCTestExpectation * applyUpdateRequestExpectation2 =
+        [self expectationWithDescription:@"handleApplyUpdateRequestForNodeID called for device2"];
+
+    XCTestExpectation * notifyUpdateAppliedExpectation1 =
+        [self expectationWithDescription:@"handleNotifyUpdateAppliedForNodeID called for device1"];
+    XCTestExpectation * notifyUpdateAppliedExpectation2 =
+        [self expectationWithDescription:@"handleNotifyUpdateAppliedForNodeID called for device2"];
+
+    __block XCTestExpectation * announceResponseExpectation2;
+
+    NSString * imagePath1 = [otaRawImagePath1 stringByReplacingOccurrencesOfString:@"raw-image" withString:@"image"];
+
+    NSString * imagePath2 = [otaRawImagePath2 stringByReplacingOccurrencesOfString:@"raw-image" withString:@"image"];
+
+    // Find the right absolute path to our ota_image_tool.py script.  PWD should
+    // point to our src/darwin/Framework, while the script is in
+    // src/app/ota_image_tool.py.
+    NSString * pwd = [[NSProcessInfo processInfo] environment][@"PWD"];
+    NSString * imageToolPath = [NSString
+        pathWithComponents:@[ [pwd substringToIndex:(pwd.length - @"darwin/Framework".length)], @"app", @"ota_image_tool.py" ]];
+
+    NSTask * task = [[NSTask alloc] init];
```

Comment:
This should be task1 and what is currently "task1" should be "task2", right?

Reply:
done.